### ISSUE TITLE
Do not call the /hero/ endpoint when AMO homepage is launched on Android

### ISSUE THREAD

**Issue by AlexandraMoga** [mozilla/addons-frontend#8513]
Follow up for #8478 

### Describe the problem and steps to reproduce it:

1. Open AMO -dev homepage on android - https://addons-dev.allizom.org/en-US/android/
2. Navigate to Extensions
3. Refresh the page
4. Open your web console
5. Navigate back to the homepage by clicking on Explore or the home button
6. Check the calls made for the `/api/v4/hero/` endpoint

### What happened?
The `/hero/` endpoint is called when the request is coming from `/android/`, although the new homepage content won't be available on Android
### What did you expect to happen?
It needs to be decided if this behavior should be left as it is, or if any changes are needed

#### Update:

We decided that we should prevent this extra fetch on Android if possible.

### Anything else we should know?
from @bobsilverberg 
>It's not necessary, but the same code is used for the home page on both desktop and Android. I think it would make the code more complex to eliminate the call to the hero API on Android, and it wouldn't save much in terms of bandwidth, but I can check with the others on that. Perhaps you could also open an issue about this and we can discuss it in that issue?

**Comment by willdurand** [comment]
I think it should be left as is. We load other data for android when client app is firefox (e.g., the categories) and vice versa.

**Comment by bobsilverberg** [comment]
Hmm, I'm pretty sure that @muffinresearch said that he thinks we should do this, but I don't see a comment from him here. I'm adding this because @willdurand [said that he didn't think we should do this](https://github.com/mozilla/addons-frontend/issues/8513#issuecomment-525627606). I'd like to come to a consensus before starting to work on this, so @muffinresearch and @kumar303, could you please add your thoughts?

**Comment by willdurand** [comment]
I think the consensus was that it'd be a nice to have because if we can avoid loading data we don't need, that's always better. That does not only apply to this particular issue :)

**Comment by kumar303** [comment]
It would add some extra complexity (as noted) so I'm just not sure it's worth it. It seems like a good idea to prevent any unneeded data fetches.

**Comment by bobsilverberg** [comment]
> It would add some extra complexity (as noted) so I'm just not sure it's worth it. It seems like a good idea to prevent any unneeded data fetches.

Yes, my biggest concern is that it will add quite a bit of complexity, and I suppose it's possible that in the future we may want some of the hero data on Android, but it sounds like the consensus is that this would be a good change to make despite that.

**Comment by muffinresearch** [comment]
We do need to think that for mobile users data can be expensive, so anything that minimizes wasted bytes over the wire is something we should aim for.

The value vs complexity idea probably needs some further thought/estimation though - could this be generalized in a way that makes it easy to only do a data request on desktop for a given API call? Thus fixing this case and any others where there's a similar issue such as the Categories case that Will mentioned.

**Comment by bobsilverberg** [comment]
> We do need to think that for mobile users data can be expensive, so anything that minimizes wasted bytes over the wire is something we should aim for.
> 
> The value vs complexity idea probably needs some further thought/estimation though - could this be generalized in a way that makes it easy to only do a data request on desktop for a given API call? Thus fixing this case and any others where there's a similar issue such as the Categories case that Will mentioned.

That does seem worth thinking about. In this case it's an entire API call that we're either issuing or not, whereas in some cases it could be *different* API calls that might be issued on desktop vs. android, so I'm not sure how easy it would be to generalize it. The idea of making `clientApp` available to sagas, which could then use it to (potentially) decide which APIs to call, and which specific calls to make, does seem like an interesting one.

One other thing that complicates this is the way sagas are coupled to reducers in that a reducer action which dispatches an API call (via a saga) usually has a corresponding action that loads the data. If we start receiving different data from sagas depending on `clientApp`, then we either need different corresponding load actions, or smart actions which understand that they're loading different data based on `clientApp`, either of which further increase the complexity compared with what we have now.

One final thing that comes to mind while we're discussing this is the fact that `clientApp` could possibly be a thing of the past at some point (as @muffinresearch has mentioned to me), and at that point all of this potentially becomes moot, unless we start using user agent to identify mobile users instead.

**Comment by bobsilverberg** [comment]
After some discussion among the engineering team, it was decided not to pursue this as the complexity outweighs the benefit.

**Comment by Bostyan** [comment]
What I like to find is how removed Firefox version view for websites? At home on all computers is Firefox 52 if only was group of devs and keep Firefox up to date with old addons will be awesome!! Browsers are really bad developed... they do not have anything useful for better and faster browsing. No sidebar, no multi-row, no Tab Groups... Using additional window to see pages names are we in stone age? Sidebar in new Firefox I will not even talk....

If my father didn`t have multi-row oh good browsing will be pain..... he use or leave open 50 or more tabs. 100 tabs use less then 1 GB ram.... and on 24" screen is not a problem at all with any i7 gen.

I started using Firefox only because of Tab Groups and Multi-row. I still like Opera 12 with sidebar and when I found All-in-One Sidebar sure in version 3.0+ sidebar was not like it is AIO. "Split" browser in 50/50 one site in sidebar and other in normal tab just awesome. For this are only two clicks :) Window mode in year 2020 is just sad really sad....


Anyway anyone know how I change or disable version?
